scripts/simplify_network.py

```diff
@@ -522,6 +523,91 @@ def consense(x):
     return clustering.network, clustering.busmap
 
 
+def drop_isolated_nodes(n, drop_e_threshold, n_periods_in_year=8760):
+    generators_sum_origin = n.generators.p_nom.sum()
+    load_sum_origin = n.loads_t.p_set.sum().sum()
+
+    loads_t_ann = n_periods_in_year * n.loads_t.p_set.mean(axis=0)
+
+    # duplicated sub-networks mean that there is at least one interconnection between buses
+    i_islands = n.buses[
+        (~n.buses.duplicated(subset=["sub_network"], keep=False))
+        & (n.buses.carrier == "AC")
+    ].index
+    i_islands_drop = i_islands[loads_t_ann[i_islands] <= drop_e_threshold]
+
+    i_generators_drop = n.generators[n.generators.bus.isin(i_islands_drop)].index
+
+    n.mremove("Bus", i_islands_drop)
+    n.mremove("Load", i_islands_drop)
```

Comment:
For loads please verify the name of the loads to be removed, the index may be different than the index of buses.
You can quickly check by looking at the demand profile csv

Reply:
Fixed. Thank you :)